torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1164,10 +1164,40 @@ def _register_param_handles_from_handles(
         for flat_param in self.params:
             self._init_param_attributes(flat_param)
         self._register_flat_params()
+        self._construct_module_to_handles()
         # Re-register the forward hooks since the handle construction changed
         self._register_pre_forward_hooks()
         self._register_post_forward_hooks()
 
+    def _construct_module_to_handles(self) -> None:
```

Comment:
Let me think about this method more tomorrow, but to clarify, is this method meant to fully initialize `_module_to_handles` in each call? Or, for example, on the second call, it is meant to replace some entries but keep the others?

If it is meant to fully initialize `_module_to_handles` in each call, should we include
```
self._module_to_handles.clear()
```
to begin the method for robustness (since the method modifies the data structure in place)?

Reply:
done